### PR TITLE
remove user from channel controller

### PR DESCRIPTION
**Pull request by WadadParker**
Issue: #48 

Created the controller function to remove user from channel as instructed in issue requirements with proper validation and relevant error responses

### CODE REVIEW

**Comment by automatefreely** [comment]
Have you read this lines from issue?
image


Modify the code, donot put unnecessary checks, also userId is not provided in param rather in body in form of user object of mongoose

**Comment by WadadParker** [comment]
I had seen in api.md that /:id was in the route, so got confused. I have made the changes

File: src/controllers/channel.controller.js [hunk 1]
```diff
@@ -93,4 +94,25 @@ exports.getChannels = async (req, res)=>{
...
+    try {
+        const{user,channel} = req.body;
+
+        const filteredChannel = channel.members.filter(user=>user===user._id);
+        const filteredUser = User.channels.filter(({channel})=>channel===channel._id);
+
+        const updatedChannel = await Channel.findByIdAndUpdate(channel._id,{members:filteredChannel});
+        const updatedUser = await User.findByIdAndUpdate(user._id,{channels:filteredUser});
+
+        return response_200(res,"User Removed from Channel Successfully")
```
Comment (automatefreely) [inline src/controllers/channel.controller.js]: Where do you remove the user. 
Also read issue description agin there are two users involved one user is adding other is being added

File: src/controllers/channel.controller.js [hunk 2]
```diff
@@ -140,4 +140,33 @@ exports.AddUserToChannel = async (req, res) => {
...
+            return response_400(res, "User is not a part of the channel");
+        }
+
+        const userToBeRemoved = await User.findById(userToBeRemovedId);
+
+        const filteredChannel = channel.members.filter(user=>user===userToBeRemovedId);
+        const filteredUser = userToBeRemoved.channels.filter(({channel})=>channel===channel._id);
+
+        const updatedChannel = await Channel.findByIdAndUpdate(channel._id,{members:filteredChannel});
+        const updatedUser = await User.findByIdAndUpdate(userToBeRemovedId,{channels:filteredUser});
```
Comment (automatefreely) [inline src/controllers/channel.controller.js]: Explain the logic here

File: src/controllers/channel.controller.js [hunk 3]
```diff
@@ -140,4 +140,33 @@ exports.AddUserToChannel = async (req, res) => {
     {
         return response_500(res, "Error removing user from channel", err);
     }
+}
+
+exports.removeUserFromChannel = async (req,res)=>
+{
+    try {
+        const{user,channel} = req.body;
+        const userToBeRemovedId = req.params.id;
```
Comment (automatefreely) [inline src/controllers/channel.controller.js]: req.params.id      ->   req.body.userId

File: src/controllers/channel.controller.js (hunk 2 above)
Comment (WadadParker) [reply src/controllers/channel.controller.js]: https://discord.com/channels/885149696249708635/1183104453344252014/1186623169805680671

File: src/controllers/channel.controller.js (hunk 3 above)
Comment (WadadParker) [reply src/controllers/channel.controller.js]: But won't that be the user calling the API? The admin? Or we are having a middleware which fetches the user to be removed for us and passes it in req.body?

**Pull request merged**